nomad/state/state_store.go

```diff
@@ -1894,6 +1894,80 @@ func (s *StateStore) DeleteJobTxn(index uint64, namespace, jobID string, txn Txn
 		return err
 	}
 
+	// Delete job deployments
+	deployments, err := s.DeploymentsByJobID(nil, namespace, job.ID, true)
+	if err != nil {
+		return fmt.Errorf("deployment lookup for job %s failed: %v", job.ID, err)
+	}
+
+	for _, deployment := range deployments {
+		// Lookup the deployment
+		existing, err := txn.First("deployment", "id", deployment.ID)
+		if err != nil {
+			return fmt.Errorf("deployment lookup failed: %v", err)
+		}
```

Comment:
Why do we need to lookup the deployment? Can `DeploymentsByJobID()` return a deployment that doesn't exist?